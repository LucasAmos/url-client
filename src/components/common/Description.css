.description{
  margin: 50px 0px 50px 0px;
}

.description .col-one {
  text-align: center;
}

.description .col-one h2 {
  font-size: 40px;
  font-family: 'Roboto Condensed';
  font-weight: 500;
}

.description .col-two {
  margin: 20px 0px 0px 0px;
  text-align: center;
}

.description .linkedin-button {
  background-color: #fff;
  color: black;
  border: none;
  font-size: 20px;
  padding: 5px 20px 5px 20px;
  border: 2px solid black !important;
}

.description .linkedin-button:hover {
  background-color: rgba(0, 0, 0, 0.1);
}

.description .linkedin-button:active {
  background-color: #283e4a3a;
}

@media (min-width: 1200px) {
  .description .col-one h2 {
    font-size: 50px;
  }
}

@media (max-width: 500px) {
  .description {
    margin-top: 40px;
    margin-bottom: 40px;
  }
}

