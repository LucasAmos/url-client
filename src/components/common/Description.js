import React from 'react';
import {
  Container, Row, Col, Button,
} from 'react-bootstrap';
import Fade from 'react-reveal/Fade';
import './Description.css';

export default function Description() {
  return (
    <div className="description">
      <Container>
        <Row>
          <Col lg={12} className="col-one">
            <Fade>
              <h2>
                FULL STACK SOFTWARE ENGINEER
              </h2>
            </Fade>
          </Col>
          <Col lg={12} className="col-two">
            <a
              href="https://www.linkedin.com/in/lucasamos/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Fade delay={200}>
                <Button
                  variant="light"
                  className="linkedin-button hvr-pulse-grow"
                >
                  LinkedIn
                </Button>
              </Fade>
            </a>
          </Col>
        </Row>
      </Container>
    </div>
  );
}
