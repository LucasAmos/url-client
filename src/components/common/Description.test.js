import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Description from './Description';

describe('Description component', () => {
  let savedError;
  beforeEach(() => {
    savedError = console.error;
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = savedError;
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      shallow(
        <Description />,
      ),
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Description />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
