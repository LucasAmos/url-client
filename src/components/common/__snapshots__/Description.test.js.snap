// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component matches the snapshot 1`] = `
<div
  className="description"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-one col-lg-12"
      >
        <h2
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          FULL STACK SOFTWARE ENGINEER
        </h2>
      </div>
      <div
        className="col-two col-lg-12"
      >
        <a
          href="https://www.linkedin.com/in/lucasamos/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="react-reveal"
            style={
              Object {
                "opacity": undefined,
              }
            }
          >
            <button
              className="linkedin-button hvr-pulse-grow btn btn-light"
              disabled={false}
              type="button"
            >
              LinkedIn
            </button>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
